multiaccounts/settings/database.go

```diff
@@ -254,8 +254,22 @@ func (db *Database) saveSetting(setting SettingField, value interface{}) error {
 	return nil
 }
 
+func (db *Database) castValue(sf SettingField, value interface{}) (interface{}, error) {
+	// Additional cast for int64 values, which are casted to float64 when unmarshalling from JSON
+	switch sf.reactFieldName {
+	case ReactFieldDisplayAssetsBelowBalanceThreshold:
```

Comment:
Can we bubble up this responsibility of dealing with the type of a specific setting field from this generic value reading to its setting? Looking at the implementation the best will be to add an optional `srcValueCastHandler` member to `SettingField`, similar to `SettingsField.valueHandler`. This way we can configure the cast handler only for this specific field.